src/page-modules/assistant/layout.tsx

```diff
@@ -35,23 +34,28 @@ function AssistantLayout({ children, tripQuery }: AssistantLayoutProps) {
   const [searchTime, setSearchTime] = useState<SearchTime>(
     tripQuery.searchTime,
   );
-  const [transportModeFilter, setTransportModeFilter] = useState(
-    getInitialTransportModeFilter(tripQuery.transportModeFilter),
-  );
   const [isSwapping, setIsSwapping] = useState(false);
   const [isPerformingSearchNavigation, setIsPerformingSearchNavigation] =
     useState(false);
   const [geolocationError, setGeolocationError] = useState<string | null>(null);
 
-  const setValuesWithLoading = async (override: Partial<FromToTripQuery>) => {
-    const query = createTripQuery(
-      {
-        ...tripQuery,
-        ...override,
-        searchTime,
-      },
-      transportModeFilter,
-    );
+  // Loading the transport mode filter data here instead of in the component
+  // avoids the data loading when the filter is mounted which causes the
+  // height to be incorrect and the animation to be janky.
+  const { data } = useSWRImmutable(
```

Comment:
Could consider changing this to `transportModeFilter` to be more explicit

Reply:
Updated